internal/controller/streams.go

```diff
@@ -17,21 +18,24 @@ import (
 )
 
 type streamClient struct {
-	conn                      *grpc.ClientConn
-	streamKubernetesResources pb.KubernetesInfoService_SendKubernetesResourcesClient
-	streamKubernetesFlows     pb.KubernetesInfoService_SendKubernetesNetworkFlowsClient
-	logStream                 pb.KubernetesInfoService_SendLogsClient
+	ciliumNamespace          string
+	ciliumNetworkFlowsStream pb.KubernetesInfoService_SendKubernetesNetworkFlowsClient
+	conn                     *grpc.ClientConn
+	client                   pb.KubernetesInfoServiceClient
+	logStream                pb.KubernetesInfoService_SendLogsClient
+	resourceStream           pb.KubernetesInfoService_SendKubernetesResourcesClient
```

Comment:
Are those ever set to nil again when a stream is closed?
Why not? Why keep a reference to a closed stream?

Reply:
Ideally the log and resource stream should never be closed right? Both should be always connected or attempting to connect. I can set the ciliumStream to nil though